kivy/tests/test_uix_carousel.py

```diff
@@ -1,15 +1,73 @@
 import pytest
 
 
-def test_remove_widget():
+@pytest.fixture(
+    scope='session', params=(True, False), ids=lambda v: 'loop=' + str(v))
+def loop(request):
+    return request.param
+
+
+def test_remove_widget(capsys, loop):
     from kivy.uix.carousel import Carousel
     from kivy.uix.widget import Widget
 
-    c = Carousel()
-    w = Widget()
-    c.add_widget(w)
+    c = Carousel(loop=loop)
+    assert c.index is None
+    assert c.current_slide is None
+    assert len(c.children) == 0
+    assert len(c.slides) == 0
+
+    N_SLIDES = 4
+    for i in range(N_SLIDES):
+        c.add_widget(Widget())
+    assert c.index == 0
+    assert c.current_slide == c.slides[0]
+    assert len(c.children) == 3 if loop else 2
+    assert len(c.slides) == N_SLIDES
+
+    # remove a slide(bigger index than the current_slide's).
+    c.index = 2
+    old_index = c.index
+    old_current_slide = c.current_slide
+    c.remove_widget(c.slides[3])
+    assert c.index == old_index
+    assert c.current_slide is old_current_slide
+    assert len(c.children) == 3 if loop else 2
+    assert len(c.slides) == 3
+
+    # remove a slide(smaller index than the current_slide's).
+    old_index = c.index
+    old_current_slide = c.current_slide
+    c.remove_widget(c.slides[1])
+    assert c.index == old_index - 1
+    assert c.current_slide is old_current_slide
+    assert len(c.children) == 2
+    assert len(c.slides) == 2
+
+    # remove the current_slide.
+    # This is where the conditions mentioned in issue #6370 is met.
+    old_index = c.index
+    old_current_slide = c.current_slide
+    c.remove_widget(c.current_slide)
+    assert c.index == old_index - 1
+    assert c.current_slide is not old_current_slide
     assert len(c.children) == 1
     assert len(c.slides) == 1
-    c.remove_widget(w)
+
+    # remove the current_slide(the last one left).
+    c.remove_widget(c.current_slide)
+    assert c.index is None
+    assert c.current_slide is None
     assert len(c.children) == 0
     assert len(c.slides) == 0
+
+    # check if there is no `IndexError` in stderr. See issue #6370.
+    stderr_text = capsys.readouterr().err
+    for line in stderr_text.split('\n'):
+        assert not line.startswith('IndexError')
```

Comment:
https://github.com/kivy/kivy/pull/6373 will fix so that it'll actually raise an exception and so that you won't need to hunt in the sdterr. Can you rebase on that once I merge it (when the my PR's test pass)?

Additionally, for the future, instead of `startswith` which is a little unreliable in case there are beginning spaces etc., using a regex match which is tolerant to that would have been better.

Reply:
Thank you. I updated the PR.

> Additionally, for the future, instead of `startswith`...

Yes I agree with that. I was writing `stderr_text.find('IndexError')` at first, but it was too slow. That's why.